elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -252,56 +251,156 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
-            if (!measures.containsKey(m.getName())) {
-                measures.put(m.getName(), m);
-            }
-        });
+        Inheritance action = () -> {
+                table.getMeasures().forEach(measure -> {
+                    if (!measures.containsKey(measure.getName())) {
+                        measures.put(measure.getName(), measure);
+                    }
+                });
+                return measures;
+        };
+
+        action.inherit();
         if (table.hasParent()) {
-            getInheritedMeasures(table.getParent(this.elideTableConfig), measures);
+            getInheritedMeasures(table.getParent(elideTableConfig), measures);
         }
         return measures;
     }
 
     private Map<String, Dimension> getInheritedDimensions(Table table, Map<String, Dimension> dimensions) {
-        table.getDimensions().forEach(dim -> {
-            if (!dimensions.containsKey(dim.getName())) {
-                dimensions.put(dim.getName(), dim);
-            }
-        });
+        Inheritance action = () -> {
+            table.getDimensions().forEach(dimension -> {
+                if (!dimensions.containsKey(dimension.getName())) {
+                    dimensions.put(dimension.getName(), dimension);
+                }
+            });
+            return dimensions;
+        };
+        action.inherit();
         if (table.hasParent()) {
-            getInheritedDimensions(table.getParent(this.elideTableConfig), dimensions);
+            getInheritedDimensions(table.getParent(elideTableConfig), dimensions);
         }
         return dimensions;
     }
 
     private Map<String, Join> getInheritedJoins(Table table, Map<String, Join> joins) {
-        table.getJoins().forEach(join -> {
-            if (!joins.containsKey(join.getName())) {
-                joins.put(join.getName(), join);
-            }
-        });
+        Inheritance action = () -> {
+            table.getJoins().forEach(join -> {
+                if (!joins.containsKey(join.getName())) {
+                    joins.put(join.getName(), join);
+                }
+            });
+            return joins;
+        };
+        action.inherit();
         if (table.hasParent()) {
-            getInheritedJoins(table.getParent(this.elideTableConfig), joins);
+            getInheritedJoins(table.getParent(elideTableConfig), joins);
         }
         return joins;
     }
 
+    private String getInheritedSchema(Table table) {
+        Inheritance action = () -> {
+            return table.getSchema();
+        };
+
+        String schema = (String) action.inherit();
+
+        if (schema == null && table.hasParent()) {
+            schema = getInheritedSchema(table.getParent(elideTableConfig));
+        }
+        return schema;
+    }
+
+    private String getInheritedConnection(Table table) {
+        Inheritance action = () -> {
+            return table.getDbConnectionName();
+        };
+
+        String conn = (String) action.inherit();
```

Comment:
Lines 371-77 and below could be consolidated into a getInheritedProperty function that all the other property functions could leverage.